app/models/upload.rb

```diff
@@ -123,14 +123,21 @@ def short_url
     "upload://#{short_url_basename}"
   end
 
-  def short_path
-    self.class.short_path(sha1: self.sha1, extension: self.extension)
+  def uploaded_before_secure_media_enabled?
```

Comment:
Why is this function needed? Don't we backfill the uploads table when secure media is enabled?

Reply:
@davidtaylorhq you don't _have_ to run the rake task that goes and marks old uploads as secure, and it is not automatic. Depending on the forum it may not even be wise if there are a lot of uploads reused all over the place, because the first time it is used in a post that post will be used as the access control post, which may break subsequent uses of the upload.

You can just turn on secure uploads + have it apply from that point in time. From an issue we had internally this check on pull hotlinked is used so if someone copies an old non-secure image into a secure context it is always created as a new upload so it does not get an access_control_post attached to the old upload.